packages/headless/src/controllers/facets/automatic-facet-builder/headless-automatic-facet-builder.ts

```diff
@@ -0,0 +1,95 @@
+import {configuration} from '../../../app/common-reducers';
+import {CoreEngine} from '../../../app/engine';
+import {SearchEngine} from '../../../app/search-engine/search-engine';
+import {setDesiredCount} from '../../../features/facets/automatic-facet-set/automatic-facet-set-actions';
+import {automaticFacetSetReducer as automaticFacetSet} from '../../../features/facets/automatic-facet-set/automatic-facet-set-slice';
+import {searchReducer as search} from '../../../features/search/search-slice';
+import {
+  AutomaticFacetSection,
+  ConfigurationSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  Controller,
+  buildController,
+} from '../../controller/headless-controller';
+import {
+  AutomaticFacet,
+  buildAutomaticFacet,
+} from '../automatic-facet/headless-automatic-facet';
+
+export interface AutomaticFacetBuilder extends Controller {
+  /**
+   * The state of the `AutomaticFacetBuilder` controller.
+   */
+  state: AutomaticFacetBuilderState;
+}
+
+export interface AutomaticFacetBuilderState {
+  /**
+   * @beta - This property is part of the automatic facets feature.
+   * Automatic facets are currently in beta testing and should be available soon.
+   *
+   * The list of automatic facet controllers.
+   */
+  automaticFacets?: AutomaticFacet[];
```

Comment:
If my other comments are correct, otherwise ignore 😅 
```suggestion
  automaticFacets: AutomaticFacet[];
```